### PR TITLE
Deprecate the prefix property of azurerm_cosmosdb_account since API's behavior has been changed

### PR DESCRIPTION
**Pull request by neil-yechenwei**
fixes terraform-providers/terraform-provider-azurerm#5347

### CODE REVIEW

File: azurerm/internal/services/cosmos/cosmosdb_account_resource.go [hunk 1]
```diff
@@ -687,6 +685,7 @@ func resourceArmCosmosDbAccountApiUpsert(client *documentdb.DatabaseAccountsClie
 	stateConf := &resource.StateChangeConf{
 		Pending:    []string{"Creating", "Updating", "Deleting", "Initializing"},
 		Target:     []string{"Succeeded"},
+		Timeout:    180 * time.Minute,
```
Comment (tombuildsstuff) [inline azurerm/internal/services/cosmos/cosmosdb_account_resource.go:688]: this should be using the user specified timeout - rather than a hard-coded one

**Comment by marc-sensenich** [comment]
@katbyte and @neil-yechenwei - I started https://github.com/terraform-providers/terraform-provider-azurerm/pull/7597 for the Cosmos DB API update to the latest provided version by the Azure SDK for Go with the intentions of working on #6164 afterwards. I did a search after my PR and noticed this PR was also open

File: azurerm/internal/services/cosmos/cosmosdb_account_resource.go (hunk 1 above)
Comment (neil-yechenwei) [reply azurerm/internal/services/cosmos/cosmosdb_account_resource.go]: updated

**Comment by neil-yechenwei** [comment]
@katbyte & @marc-sensenich , I think I can update this PR once #6253/#7597 got merged.

**Pull request closed without merging**